app/Models/Forum/Topic.php

```diff
@@ -313,12 +341,24 @@ public function scopeRecent($query, $params = null)
 
     public function nthPost($n)
     {
-        return $this->posts()->skip(intval($n) - 1)->first();
+        $posts = $this->posts();
+
+        if (priv_check('ForumTopicModerate')->can()) {
+            $posts->withTrashed();
+        }
+
+        return $posts->skip(intval($n) - 1)->first();
     }
 
     public function postPosition($postId)
     {
-        return $this->posts()->where('post_id', '<=', $postId)->count();
+        $posts = $this->posts();
+
+        if (priv_check('ForumTopicModerate')->can()) {
+            $posts->withTrashed();
+        }
+
+        return $posts->where('post_id', '<=', $postId)->count();
     }
 
     public function postsPosition($sortedPosts)
```

Comment:
this probably can be removed to save passing around the `$withTrashed` variable (just increment the first post's `postPosition` in the view).